internal/btpmanager/credentials/utils.go

```diff
@@ -0,0 +1,22 @@
+package btpmgrcreds
+
+import (
+	"fmt"
+	"time"
+)
+
+func CallWithRetry[T any](clousure func() (T, error), maxAttempts int, nextLogicCallWait time.Duration) (T, error) {
+	doneAttempts := 0
+	for {
+		result, err := clousure()
+		if err == nil {
+			return result, nil
+		}
+		doneAttempts++
+		if doneAttempts >= maxAttempts {
+			var empty T
+			return empty, fmt.Errorf("%s. (error still occured after %d retries)", err, maxAttempts)
```

Comment:
```suggestion
			return empty, fmt.Errorf("%s. (error still occurred after %d retries)", err, maxAttempts)
```
The thing is that the error not occurred after `n` retries but during `n` retries

Reply:
Suggestions by Iwona:
```
N retries returned error/ Error occurred in n retries/ All n retries resulted in error
```